lighthouse-core/audits/consistently-interactive.js

```diff
@@ -0,0 +1,247 @@
+          end: longTasks[index + 1].start + navStartTsInMs,
+        });
+      }
+    });
+
+    return quietPeriods;
+  }
+
+  /**
+   * @param {!Array<{start: number, end: number}>} longTasks
+   * @param {{timestamps: {navigationStart: number, firstMeaningfulPaint: number,
+   *    traceEnd: number}}} traceOfTab
+   * @return {!Object}
+   */
+  static findOverlappingQuietPeriods(longTasks, networkRecords, traceOfTab) {
+    const FMPTsInMs = traceOfTab.timestamps.firstMeaningfulPaint;
+
+    const isLongEnoughQuietPeriod = period =>
+        period.end > FMPTsInMs + REQUIRED_QUIET_WINDOW &&
+        period.end - period.start >= REQUIRED_QUIET_WINDOW;
+    const networkQuietPeriods = this._findNetworkQuietPeriods(networkRecords, traceOfTab)
+        .filter(isLongEnoughQuietPeriod);
+    const cpuQuietPeriods = this._findCPUQuietPeriods(longTasks, traceOfTab)
+        .filter(isLongEnoughQuietPeriod);
+
+    const cpuQueue = cpuQuietPeriods.slice();
+    const networkQueue = networkQuietPeriods.slice();
+
+    // We will check for a CPU quiet period contained within a Network quiet period or vice-versa
+    let cpuCandidate = cpuQueue.shift();
+    let networkCandidate = networkQueue.shift();
+    while (cpuCandidate && networkCandidate) {
+      if (cpuCandidate.start >= networkCandidate.start) {
+        // CPU starts later than network, it must be contained by network or we check the next network
+        if (networkCandidate.end >= cpuCandidate.start + REQUIRED_QUIET_WINDOW) {
+          return {
+            cpuQuietPeriod: cpuCandidate,
+            networkQuietPeriod: networkCandidate,
+            cpuQuietPeriods,
+            networkQuietPeriods,
+          };
+        } else {
+          networkCandidate = networkQueue.shift();
+        }
+      } else {
+        // Network starts later than CPU, it must be contained by CPU or we check the next CPU
+        if (cpuCandidate.end >= networkCandidate.start + REQUIRED_QUIET_WINDOW) {
+          return {
+            cpuQuietPeriod: cpuCandidate,
+            networkQuietPeriod: networkCandidate,
+            cpuQuietPeriods,
+            networkQuietPeriods,
+          };
+        } else {
+          cpuCandidate = cpuQueue.shift();
+        }
+      }
+    }
+
+    const culprit = cpuCandidate ? 'Network' : 'CPU';
+    throw new Error(`${culprit} did not quiet for at least 5s before the end of the trace.`);
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!Promise<!AuditResult>}
+   */
+  static audit(artifacts) {
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const networkRecords = artifacts.networkRecords[Audit.DEFAULT_PASS];
+    const computedTraceArtifacts = [
+      artifacts.requestTracingModel(trace),
+      artifacts.requestTraceOfTab(trace),
+    ];
+
+    return Promise.all(computedTraceArtifacts)
+      .then(([traceModel, traceOfTab]) => {
+        if (!traceOfTab.timestamps.firstMeaningfulPaint) {
```

Comment:
(though in the spec I'm not actually sure this is true, it just seemed like the proper thing to do. I should raise it)